Automattic/woocommerce-payments
# Fix issue where order status and data are not updated when an order is in the processing state and being captured.

Fixes #5671

#### Changes proposed in this Pull Request

This PR addresses an issue where, if an order is in a processing state (Processing or Completed) and the transaction is uncaptured, capturing the transaction or order fails to update order metadata such as intent status and order notes. 

The issue occurs because the previous method skipped updating the intent status when the order was in the 'processing' state. To resolve this, this PR introduces a new function that bypasses this check and leverages existing functions to properly process the order, ensuring that all metadata and states are updated correctly.

#### Testing Instructions  

1. Enable manual capture.  
2. Switch to the `develop` branch or any branch without the code from this PR.  
3. Place an order and complete the payment.  
4. In the admin panel, locate the order and confirm it is in the "On-hold" state.  
5. Change the order status to "Processing."  
6. Navigate to **Payments -> Transactions**, then open the "Uncaptured Transactions" tab.  
7. Find the newly created order and click the **Capture** button.  
   - Observe that the capture note is missing on the order screen.  
   - Check Stripe and verify the transaction is captured.  
8. Attempt to refund the order:  
   - Notice that you cannot refund because the order remains in the "requires capture" state, and an error message appears.  

9. Switch to this branch.  
10. Repeat steps 1–6.  
11. After capturing the transaction:  
    - Verify that the order screen now includes a note indicating the capture.  
    - Attempt to refund the order and confirm that refunds work as expected.  

-------------------


**Post merge**
